src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1482,13 +1486,15 @@ static void query_for_backends_locked(grpc_exec_ctx* exec_ctx,
   GPR_ASSERT(glb_policy->lb_channel != NULL);
   if (glb_policy->shutting_down) return;
 
+  if (glb_policy->retry_timer_active) {
```

Comment:
What problem does this fix?  In what case can we get here when the retry timer is active?

If this fix is unrelated to the `client_stats` fix, it might be better to move it to a separate PR.

Reply:
This:

```
I1109 09:40:39.169257  994714 grpclb.cc:1495] [grpclb 0x7f1dbb048300] Query for backends (lb_channel: 0x7f1dba165000, lb_call: 0x7f1db5480020)
...
I1109 09:40:39.171471  994714 grpclb.cc:1267] [grpclb 0x7f1dbb048300] Restaring call to LB server
E1109 09:40:39.172154  994714 grpclb.cc:1269] assertion failed: glb_policy->lb_call == NULL

```
That is, the timer, having a life of its own, may fire after the LB call has already been restarted.

ok, I'll move it to another PR.